app/auth/validate.js

```diff
@@ -0,0 +1,15 @@
+const api = require('../api')
+const cache = require('../cache')
+
+const validate = async (decoded, request, h) => {
+  let isValid = false
+  const agreement = await cache.get('agreement', request.yar.id)
+
+  if (agreement?.callerId === decoded.callerId) {
+    isValid = true
```

Comment:
Could this be simplified to something like the below to avoid the reassigning of variables.

```
const agreement = await cache.get('agreement', request.yar.id)
return { isValid: agreement?callerId === decoded.callerId }
```